syncer/syncer.go

```diff
@@ -1657,11 +1717,17 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 		if firstEndPos == nil {
 			return errors.Errorf("no valid End_log_pos of the first DDL exists for sharding group with source %s", source)
 		}
+
+		allResolved, err2 := s.sgk.ResolveShardingDDL(ddlInfo.tableNames[1][0].Schema, ddlInfo.tableNames[1][0].Name)
+		if err != nil {
+			return errors.Trace(err2)
+		}
 		*ec.shardingReSyncCh <- &ShardingReSync{
```

Comment:
can we remove re-shard mechanism

Reply:
I thought we can just close and redirect the *global* stream to the right position (the pos after the sharding DDL), and no *sharding* stream needed.

@amyangfei you can consider about it.

Reply:
Firstly I just removed the usage of `sharding stream`, just redirect `global stream` to proper position when needed.
Secondly for the re-shard notification mechanism, as we have the checkpoint mechanism, so in theory, the re-shard is not needed, if we can prove the following works right:
`shard DDL resovled -> checkpoint resolved -> start shard re-sync -> syncer unit crash -> syncer unit resume -> re-sync from checkpoint -> continue syncing binlog events`.
If this doesn't work well, `re-shard` mechanism does not work well either.

Reply:
yep, you're right